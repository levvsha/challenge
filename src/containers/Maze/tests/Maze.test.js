import React from 'react';
import renderer from 'react-test-renderer';

import { Maze, mapStateToProps } from '../';

const mazeMatrixMock = [
  [
    {
      left: true,
      top: true,
    },
    {
      top: true,
    },
    {
      left: true,
      top: true,
    },
  ],
  [
    {
      left: true,
      top: true,
    },
    {},
    {
      left: true,
      top: true,
    },
  ],
  [
    {
      left: true,
    },
    {
      top: true,
    },
    {
      left: true,
    },
  ],
];

describe('<Maze />', () => {
  it('mapStateToProps', () => {
    const mockState = {
      game: {
        mazeMatrix: [],
        ponyCoords: {},
        enemyCoords: {},
        exitCoords: {},
        isGameFinished: false,
        isWin: false,
        foo: 'bar'
      },
      foo: 'bar'
    };

    const expectedState = {
      mazeMatrix: [],
      ponyCoords: {},
      enemyCoords: {},
      exitCoords: {},
      isGameFinished: false,
      isWin: false,
    };

    expect(mapStateToProps(mockState)).toEqual(expectedState);
  });

  it('<Maze /> renders correctly', () => {

    const tree = renderer.create(
      <Maze
        enemyCoords={{ x: 1, y: 1 }}
        exitCoords={{ x: 1, y: 2 }}
        ponyCoords={{ x: 2, y: 2 }}
        isGameFinished={false}
        isWin={false}
        mazeMatrix={mazeMatrixMock}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
