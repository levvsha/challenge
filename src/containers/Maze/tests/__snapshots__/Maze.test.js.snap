// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Maze /> <Maze /> renders correctly 1`] = `
<div
  className="c-maze"
>
  <div />
  <div />
  <div />
  <div
    className="maze-row"
  >
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
      <div
        className="maze-left-border"
      />
    </div>
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
    </div>
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
      <div
        className="maze-left-border"
      />
    </div>
  </div>
  <div
    className="maze-row"
  >
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
      <div
        className="maze-left-border"
      />
    </div>
    <div
      className="maze-cell"
    />
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
      <div
        className="maze-left-border"
      />
    </div>
  </div>
  <div
    className="maze-row"
  >
    <div
      className="maze-cell"
    >
      <div
        className="maze-left-border"
      />
    </div>
    <div
      className="maze-cell"
    >
      <div
        className="maze-top-border"
      />
    </div>
    <div
      className="maze-cell"
    >
      <div
        className="maze-left-border"
      />
    </div>
  </div>
</div>
`;
